### PR TITLE
chore: use handle instead of globals for injection

### PR DESCRIPTION
**Pull request by jrandolf-google**
This PR replace globals injection with handle passing. This alleviates the issue of globals creep into the main world.

### CODE REVIEW

File: src/injected/injected.ts [hunk 1]
```diff
@@ -2,13 +2,12 @@ import {createDeferredPromise} from '../util/DeferredPromise.js';
...
-      ...Poller,
-      ...util,
-      createDeferredPromise,
-    },
-  })
-);
+const PuppeteerUtil = Object.freeze({
+  ...Poller,
+  ...util,
+  createDeferredPromise,
```
Comment (OrKoN) [inline src/injected/injected.ts:6]: let's not include stuff that is not used before we need it. I believe only createDeferredPromise is used?

File: src/common/IsolatedWorld.ts [hunk 2]
```diff
@@ -164,13 +168,26 @@ export class IsolatedWorld {
...
     this.#context = createDeferredPromise();
   }
 
   setContext(context: ExecutionContext): void {
-    if (this.#injected) {
-      context.evaluate(injectedSource).catch(debugError);
-    }
+    (async () => {
+      try {
+        this.#puppeteerUtil.resolve(
```
Comment (OrKoN) [inline src/common/IsolatedWorld.ts]: let's extract this in it's own method?

File: src/templates/injected.ts.tmpl [hunk 3]
```diff
@@ -1,10 +1,4 @@
...
-  const InjectedUtil: {
-    createDeferredPromise: typeof createDeferredPromise;
-  };
-}
-
-/** @internal */
+/** 
+ * @internal
+ */
 export const source = SOURCE_CODE;
```
Comment (OrKoN) [inline src/templates/injected.ts.tmpl:8]: Let's add a comment that the source code is produced by rollup and in which module format (CJS)?

File: src/injected/injected.ts (hunk 1 above)
Comment (jrandolf) [reply src/injected/injected.ts:6]: I've kept `util` since we use it for testing.

File: src/common/IsolatedWorld.ts (hunk 2 above)
Comment (jrandolf) [reply src/common/IsolatedWorld.ts]: Done.

File: src/templates/injected.ts.tmpl (hunk 3 above)
Comment (jrandolf) [reply src/templates/injected.ts.tmpl:8]: Done.

File: src/common/IsolatedWorld.ts [hunk 4]
```diff
@@ -164,20 +168,35 @@ export class IsolatedWorld {
 
   clearContext(): void {
     this.#document = undefined;
+    this.#puppeteerUtil = createDeferredPromise();
```
Comment (OrKoN) [inline src/common/IsolatedWorld.ts:171]: should we call `dispose` on the previous handle? It looks like we should

Comment (jrandolf) [reply src/common/IsolatedWorld.ts:171]: The handle is destroyed since the context is destroyed.

**Review by OrKoN — APPROVED** [review]

**Pull request merged**